ansible_collections/arista/avd/roles/eos_designs/defaults/main/default-node-type-keys.yml

```diff
@@ -25,3 +25,20 @@ default_node_type_keys:
     overlay_controller:
       type: overlay-controller
       default_evpn_role: server
+
+  "mpls-vpn":
+    p:
+      type: p
+      default_evpn_role: none
```

Comment:
The template code becomes very messy with this reuse of evpn role. Also this would make it super hard to build a gateway which may be client on EVPN but server on mpls side etc.
To make this maintainable, I think we should add another key like `default_mpls_role` or `default_overlay_routing_protocol_role`

Reply:
Made new feature toggle default_mpls_overlay_role and deprecated reuse of evpn_role for mpls-vpn design.